src/java.base/share/classes/java/lang/reflect/code/op/CoreOp.java

```diff
@@ -2168,7 +2168,7 @@ public VarOp transform(CopyContext cc, OpTransformer ot) {
         VarOp(String varName, TypeElement type, Value init) {
             super(NAME, List.of(init));
 
-            this.varName = varName;
+            this.varName = varName == null || varName.isEmpty() ? null : varName;
```

Comment:
Was there is a reason why you check for an empty string?

Reply:
The compiler code may pass an empty string for the name of the Var, e.g. unnamed pattern variable.

Reply:
I think we should address that. Up to you if you wanna fix that in this PR or do it in a follow on PR.

Reply:
I think it's better to address it in a follow PR.

Reply:
Ok!